backend/rorapp/views/waitlist_entry.py

```diff
@@ -1,12 +1,24 @@
 from rest_framework import viewsets
+from rest_framework import status
 from rorapp.models import WaitlistEntry
 from rorapp.serializers import WaitlistEntryCreateSerializer
 from rest_framework.exceptions import MethodNotAllowed
+from rest_framework.response import Response
 
 class WaitlistEntryViewSet(viewsets.ModelViewSet):
     queryset = WaitlistEntry.objects.all()
     serializer_class = WaitlistEntryCreateSerializer
     
+    def create(self, request, *args, **kwargs):
+        data = request.data
+
+        isEntryPresent = WaitlistEntry.objects.filter(email=data["email"])
+
+        if isEntryPresent:
+            return Response({}, status=status.HTTP_200_OK)
```

Comment:
The problem with this solution is that a consumer of the API can still tell whether a new record has been created, or the record already exists. I appreciate that for anyone using the UI this solution is good, but it is also possible for anyone to send requests to the open API endpoint at https://api.roronline.com/api/waitlist_entry/ instead of using the UI, and if they do that they will see the status code in the response.

Instead of a 200, let's send a 201, so the API pretends to have created a new record:

```suggestion
            return Response({}, status=status.HTTP_201_CREATED)
```

Reply:
Thinking about it a bit more, I realise this wouldn't be enough. You will also need to make the actual response data (see the `{}` thing) look exactly like an authentic successul creation response!

Reply:
I added the needed data that should be returned.